src/extensionFunctions.ts

```diff
@@ -898,3 +903,54 @@ export async function copyFileTextToClipboard(uri: vscode.Uri) : Promise<void> {
 		}
 	}
 } 
+
+export async function pasteClipboardToActiveEditor() : Promise<void> {
+	let currentEditor : vscode.TextEditor | undefined = vscode.window.activeTextEditor;
+	if (!currentEditor) {
+		await vscode.commands.executeCommand('workbench.action.openPreviousRecentlyUsedEditor');
+		currentEditor = vscode.window.activeTextEditor;
+	}
+	if (currentEditor) {
+		const textFromClipboard = await vscode.env.clipboard.readText();
+		if (!textFromClipboard) {
+			await vscode.window.showWarningMessage(`No text found on clipboard`);
+			return;
+		}
+		await currentEditor.insertSnippet(new vscode.SnippetString(textFromClipboard));
+		return;
+	}
+	await vscode.window.showWarningMessage(`No active text editor found to paste from clipboard`);
+}
+
+export async function openEditorForFile(uri: vscode.Uri) : Promise<vscode.TextEditor | undefined> {
+	for (let index = 0; index < vscode.window.visibleTextEditors.length; index++) {
+		const editor = vscode.window.visibleTextEditors[index];
+		if (editor.document.uri === uri) {
+			return editor;
+		}
+	}
+	return undefined;
+}
+
+export async function pasteClipboardToEditorForFile(uri: vscode.Uri) : Promise<void> {
+	let editorForFile : vscode.TextEditor | undefined = await openEditorForFile(uri);
+	if (!editorForFile) {
+		try {
+			await waitUntil(async () => {
+				await vscode.commands.executeCommand('vscode.open', uri, vscode.ViewColumn.Beside);
+				editorForFile = vscode.window.activeTextEditor;
+			}, EDITOR_OPENED_TIMEOUT, 1000);
+		} catch (error) {
+			await vscode.window.showWarningMessage(`No editor found for file ${uri.fsPath}. ${error}`);
+			console.log(error);
+		}
+	}
```

Comment:
I do no tunderstand why there is a conditionial wait which is trying to reopen the same file. There is no return of a boolean in the waitUntil so what is it waiting for?

Reply:
Good call